apps/dh/api-dh/source/DataHub.WebApi/GraphQL/Extensions/MarketParticipantClientExtensions.cs

```diff
@@ -62,4 +64,36 @@ internal static async Task<IEnumerable<GridAreaDto>> GetGridAreasAsync(
                     };
             });
     }
+
+    internal static async Task<IEnumerable<GridAreaDto>> GetRelevantGridAreasAsync(
+        this IMarketParticipantClient_V1 client,
+        Guid actorId,
+        Interval period,
+        CancellationToken cancellationToken = default)
+    {
+        var actor = await client.ActorGetAsync(actorId).ConfigureAwait(false);
+
+        var gridAreas = await client.GetGridAreasAsync();
```

Comment:
suggestion: fetch these in parallel for slightly faster response times

Reply:
done, every millisecond counts :D